ffi/capi/src/lib.rs

```diff
@@ -3,7 +3,10 @@
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
 #![cfg_attr(
-    all(target_os = "none", feature = "freertos"),
+    any(
+        all(feature = "freertos", not(feature = "x86tiny")),
+        all(feature = "x86tiny", not(feature = "freertos")),
```

Comment:
i wouldn't necessarily do the `all(feature, not(feature))`, dance, i'm fine with this being an error if the features are mixed (as opposed to a silent allow)

Reply:
I did it this way because `--all-features` needs to work.